client/src/components/posting-page/JobPosting.js

```diff
@@ -100,13 +119,30 @@ export class JobPosting extends Component {
     this.setState({ attachments, inputError: null })
   }
 
+  //Reads the attachment and converts it to base64
+  readFile(attachment) {
+    let reader = new FileReader()
+    let file = attachment
+    return new Promise((resolve, reject) => {
+      reader.addEventListener('load', function () {
+        resolve(this.result)
+      }, false)
+      if (file) {
+        return reader.readAsDataURL(file)
+      } else {
+        reject('foo')
```

Comment:
Maybe would be better to include a reason for why promise is rejected, i.e. the attachment is not defined